src/java.base/share/classes/java/net/URL.java

```diff
@@ -554,12 +575,19 @@ public URL(String spec) throws MalformedURLException {
      * <p>
      * For a more detailed description of URL parsing, refer to RFC2396.
      *
+     * @implSpec Parsing the URL involves calling the {@link
+     * URLStreamHandler#parseURL(URL, String, int, int) parseURL} method on the
+     * selected handler.
+     *
      * @param      context   the context in which to parse the specification.
      * @param      spec      the {@code String} to parse as a URL.
      * @throws     MalformedURLException  if no protocol is specified, or an
      *               unknown protocol is found, or {@code spec} is {@code null},
      *               or the parsed URL fails to comply with the specific syntax
-     *               of the associated protocol.
+     *               of the associated protocol. In particular, if the
+     *               underlying stream handler's {@linkplain
+     *               URLStreamHandler#parseURL parseURL method} throws
+     *               {@code IllegalArgumentException}
```

Comment:
This looks okay but it might be simpler to drop "In particular" and just say "or the stream handler's parseURL throws ...".

Reply:
Thanks @AlanBateman I have pushed all the requested changes and updated the CSR. If that looks good to you I will finalize the CSR.